src/main/java/com/google/idea/perf/methodtracer/MethodTracerController.kt

```diff
@@ -161,56 +165,47 @@ class MethodTracerController(
+                                flags
+                            )
                         }
+                        else {
+                            TracerConfig.untrace(TracePattern.ByClassPattern(classPattern))
+                        }
+                        retransformClasses(classes.toSet())
                     }
                     is TraceTarget.MethodPattern -> {
-                        val traceOption = command.traceOption
                         val className = command.target.className
                         val methodPattern = command.target.methodPattern
-                        if (traceOption != null) {
-                            if (command.enable) {
-                                TracerConfig.trace(
-                                    TracePattern.ByMethodPattern(className, methodPattern),
-                                    traceOption.tracepointFlag
-                                )
-                            }
-                            else {
-                                TracerConfig.untrace(TracePattern.ByMethodPattern(className, methodPattern))
-                            }
-                            retransformClasses(setOf(className))
+                        if (command.enable) {
+                            TracerConfig.trace(
+                                TracePattern.ByMethodPattern(className, methodPattern),
+                                flags
+                            )
                         }
+                        else {
+                            TracerConfig.untrace(TracePattern.ByMethodPattern(className, methodPattern))
+                        }
+                        retransformClasses(setOf(className))
                     }
                     is TraceTarget.Method -> {
-                        val traceOption = command.traceOption
                         val className = command.target.className
-                        val methodName = command.target.methodName
-                        val parameters = command.target.parameterIndexes
-                        if (traceOption != null && methodName != null && parameters != null) {
-                            if (command.enable) {
-                                TracerConfig.trace(
-                                    TracePattern.ByMethodName(className, methodName),
-                                    traceOption.tracepointFlag,
-                                    parameters
-                                )
-                            }
-                            else {
-                                TracerConfig.untrace(TracePattern.ByMethodName(className, methodName))
-                            }
-                            retransformClasses(setOf(className))
+                        val methodName = command.target.methodName!!
+                        val parameters = command.target.parameterIndexes!!
+                        if (command.enable) {
```

Comment:
I think I found a bug, but I'm not sure where the problem lies.  
Here is the scenario: run the following tracing command twice.
```
trace com.intellij.openapi.progress.ProgressManager#checkCanceled
```
The second time this command runs, it ends up tracing all methods in ProgressManager rather than just `checkCanceled`.

Reply:
Fixed! Based on file commit history, this bug must have existed for a while now.